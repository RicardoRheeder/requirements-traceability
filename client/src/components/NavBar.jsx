import React, { useEffect } from 'react'
import { NavLink, withRouter } from 'react-router-dom'
import { LogoutButton } from './'

export default function NavBar() {
  useEffect(() => {
    const script = document.createElement('script')

    script.src = 'assets/js/agency.js'
    script.async = true

    document.body.appendChild(script)
  }, [])

  return (
    <div className="navbar-root">
      <nav
        className="navbar navbar-dark navbar-expand-lg fixed-top bg-dark mainNav"
        id="mainNav"
      >
        <div className="container">
          <a className="navbar-brand">{'{Doc Tracer Icon}'}</a>
          <div>
            <ul className="navbar-nav text-uppercase">
              <li className="nav-item">
                <NavLink to={'/'} id="NavToHome">
                  <a className="nav-link">Home</a>
                </NavLink>
              </li>
              <li className="nav-item">
                <NavLink to={'/editor'} id="NavToEditor">
                  <a className="nav-link">Editor</a>
                </NavLink>
              </li>

              <li className="nav-item">
                <a className="nav-link">About</a>
              </li>
              <li className="nav-item">
                <a className="nav-link">Team</a>
              </li>
              <li className="nav-item">
                <a className="nav-link">Contact</a>
              </li>
              <li className="nav-item">
                <a className="nav-link">
                  <LogoutButton />
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
  )
}
