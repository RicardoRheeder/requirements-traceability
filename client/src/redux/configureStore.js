import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

// restricting extension in production using process.env.NODE_ENV which is set 'production'
import { composeWithDevTools } from 'redux-devtools-extension/logOnlyInProduction';

import store from './stores';

const enhancer = (process.env.NODE_ENV == 'development')
               ? composeWithDevTools(applyMiddleware(thunk))
               : applyMiddleware(thunk);

export default () => createStore(store, enhancer);
