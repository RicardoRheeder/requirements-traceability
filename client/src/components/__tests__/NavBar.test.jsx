import React from "react"
import Enzyme, { shallow, mount} from "enzyme"
import NavBar from "../NavBar"
import Adapter from "enzyme-adapter-react-16"
import { Provider } from "react-redux";
import configureStore from "redux-mock-store"
import { BrowserRouter as Router } from 'react-router-dom';


const mockStore = configureStore([])
Enzyme.configure({adapter: new Adapter() });

describe("Nav Bar", () => {
    let store;
    let testBar;
    beforeEach(() => {
        store = mockStore({});
    });
    test("renders", () => {
        const wrapper = shallow(<Provider store = {store}><NavBar/></Provider>);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
    test("calls navBar function to Home", () => {
        const wrapper = mount(<Router><Provider store = {store}><NavBar/></Provider></Router>);
        wrapper.update()
        wrapper.find('.nav-link').at(0).simulate('click')
        expect(wrapper).toMatchSnapshot();
    });
    test("calls navBar function to Editor", () => {
        const wrapper = mount(<Router><Provider store = {store}><NavBar/></Provider></Router>);
        wrapper.update()
        wrapper.find('.nav-link').at(1).simulate('click')
        expect(wrapper).toMatchSnapshot();
    });
});