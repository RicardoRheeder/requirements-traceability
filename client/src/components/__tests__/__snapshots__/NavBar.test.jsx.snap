// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Bar calls navBar function to Editor 1`] = `ReactWrapper {}`;

exports[`Nav Bar calls navBar function to Home 1`] = `ReactWrapper {}`;

exports[`Nav Bar renders 1`] = `ShallowWrapper {}`;
